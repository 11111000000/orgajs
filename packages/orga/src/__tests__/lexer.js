const Lexer = require('../lexer')

describe('Lexer', () => {
  it('recognize headlines', () => {
    let lexer = new Lexer()
    expect(lexer.tokenize('* TODO [#A] headline one     :swift:javascript:')).toMatchSnapshot()
  })

  it('recognize planning', () => {
    let lexer = new Lexer()
    // t = lexer.tokenize('DEADLINE: <2018-01-01 Mon>')
    // console.log(t)
    expect(lexer.tokenize('DEADLINE: <2018-01-01 Mon>')).toMatchSnapshot()
  })

})
