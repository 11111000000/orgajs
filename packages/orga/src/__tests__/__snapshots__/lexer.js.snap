// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lexer recognize headlines 1`] = `
Object {
  "content": " headline one",
  "keyword": "TODO",
  "level": 1,
  "name": "headline",
  "priority": "A",
  "tags": ":swift:javascript:",
}
`;

exports[`Lexer recognize planning 1`] = `
Object {
  "keyword": "DEADLINE",
  "name": "planning",
  "timestamp": " <2018-01-01 Mon>",
}
`;
